cart/infrastructure/defaultCartBehaviour.go

```diff
@@ -77,14 +97,16 @@ func (cob *DefaultCartBehaviour) Inject(
 		DefaultCurrency string  `inject:"config:commerce.cart.defaultCartAdapter.defaultCurrency"`
 	},
 ) {
-	cob.cartStorage = CartStorage
-	cob.productService = ProductService
-	cob.logger = Logger.WithField(flamingo.LogKeyCategory, "inmemorybehaviour")
+	cob.cartStorage = cartStorage
+	cob.productService = productService
+	cob.logger = logger.WithField(flamingo.LogKeyCategory, "inmemorybehaviour")
```

Comment:
that seems to be off, and please move the `.WithFields` to each logger call because we otherwise waste memory.